common/src/main/java/io/netty5/util/concurrent/TaskScheduler.java

```diff
@@ -0,0 +1,92 @@
+
+/**
+ * A task scheduler for {@link EventExecutor}s that want to support task scheduling.
+ *
+ * @see AbstractScheduledEventExecutor
+ */
+public interface TaskScheduler {
+
+    /**
+     * Schedule the {@link RunnableScheduledFuture} task for execution.
+     */
+    <V> Future<V> schedule(RunnableScheduledFuture<V> task);
+
+    /**
+     * Schedule the given {@link Runnable} task for execution after the given delay.
+     */
+    Future<Void> schedule(Runnable command, long delay, TimeUnit unit);
+
+    /**
+     * Schedule the given {@link Callable} task for execution after the given delay.
+     */
+    <V> Future<V> schedule(Callable<V> callable, long delay, TimeUnit unit);
+
+    /**
+     * Schedule the given {@link Runnable} task for periodic execution.
+     * The first execution will occur after the given initial delay, and the following repeated executions will occur
+     * with the given period of time between each execution is started.
+     * If the task takes longer to complete than the requested period, then the following executions will be delayed,
+     * rather than allowing multiple instances of the task to run concurrently.
+     * <p>
+     * The task will be executed repeatedly until it either fails with an exception, or its future is
+     * {@linkplain Future#cancel() cancelled}. The future thus will never complete successfully.
+     */
+    Future<Void> scheduleAtFixedRate(Runnable command, long initialDelay, long period, TimeUnit unit);
+
+    /**
+     * Schedule the given {@link Runnable} task for periodic execution.
+     * The first execution will occur after the given initial delay, and the following repeated executions will occur
+     * with the given subsequent delay between one task completing and the next task starting.
+     * The delay from the completion of one task, to the start of the next, stays unchanged regardless of how long a
+     * task takes to complete.
+     * <p>
+     * This is in contrast to {@link #scheduleAtFixedRate(Runnable, long, long, TimeUnit)} which varies the delays
+     * between the tasks in order to hit a given frequency.
+     * <p>
+     * The task will be executed repeatedly until it either fails with an exception, or its future is
+     * {@linkplain Future#cancel() cancelled}. The future thus will never complete successfully.
+     */
+    Future<Void> scheduleWithFixedDelay(Runnable command, long initialDelay, long delay, TimeUnit unit);
+
+    RunnableScheduledFuture<?> peekScheduledTask();
+
+    RunnableScheduledFuture<?> pollScheduledTask();
+
+    /**
+     * Return the task which is ready to be executed with the given {@code nanoTime}.
+     */
+    RunnableScheduledFuture<?> pollScheduledTask(long nanoTime);
+
+    void removeNextScheduledTask();
+
+    void removeScheduled(RunnableScheduledFuture<?> task);
+
+    /**
+     * Cancel all scheduled tasks.
+     */
+    void cancelScheduledTasks();
+
+    int size();
+
+    boolean isEmpty();
```

Comment:
**Suggestion**: For methods that schedule tasks (e.g., schedule, scheduleAtFixedRate, scheduleWithFixedDelay), consider specifying the behavior or exceptions when scheduling tasks in a shutdown state. This clarification can help ensure consistent usage and error handling. <br> **Code Suggestion**: 
 ```
 +    /**
 +     * Schedules the specified task for execution after the specified delay.
 +     * @param task the task to schedule
 +     * @param delay the time from now to delay execution
 +     * @param unit the time unit of the delay parameter
 +     * @return a Future representing pending completion of the task
 +     * @throws RejectedExecutionException if the task cannot be scheduled for execution
 +     * @throws IllegalStateException if the scheduler is in a shutdown state
 +     */
 +    Future<Void> schedule(Runnable command, long delay, TimeUnit unit);
 ```